catalog/plugin/catalog-plugin-metacard-validation/src/main/java/ddf/catalog/metacard/validation/MetacardValidityFilterPlugin.java

```diff
@@ -104,17 +108,20 @@ public PolicyResponse processPostQuery(Result input, Map<String, Serializable> p
         Metacard metacard = input.getMetacard();
         HashMap<String, Set<String>> securityMap = new HashMap<>();
 
-        if ((metacard.getAttribute(Validation.VALIDATION_ERRORS) != null && metacard.getAttribute(
-                Validation.VALIDATION_ERRORS)
-                .getValues() != null) || (
-                metacard.getAttribute(Validation.VALIDATION_WARNINGS) != null &&
-                        metacard.getAttribute(Validation.VALIDATION_WARNINGS)
-                                .getValues() != null)) {
+        if (filterErrors && metacard.getAttribute(Validation.VALIDATION_ERRORS) != null &&
```

Comment:
Should extract a method to check attribute ! null & attribute.getValues() ! null. That would clean up your two tests.